.github/workflows/main.yml

```diff
@@ -7,30 +7,113 @@ on:
     # run daily, this refreshes the cache
     - cron: '13 2 * * *'
 
+# Cancel a currently running workflow from the same PR, branch or tag
+# when a new workflow is triggered:
+# https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+
 jobs:
-  python-test:
-    name: Python tests
+  python2-test:
+    name: Python2 tests
     runs-on: ubuntu-22.04
-    strategy:
-      fail-fast: false
-      matrix:
-        docker-image:  # See https://hub.docker.com/_/python/tags for images
-          - python:2.7.18-alpine3.11
-          - python:3.11.4-alpine3.18
-    container: ${{ matrix.docker-image }}
+    container: python:2.7.18-alpine3.11
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
 
-      - name: Install python 2 dependencies
-        if: ${{ startsWith(matrix.docker-image, 'python:2.7.18') }}
-        run: pip install enum
+      - name: Install Python2 dependencies
+        run: pip install enum mock pytest-coverage pytest-mock
 
-      - name: Install dependencies
-        run: pip install mock pytest
+      - name: Run Python2 tests
+        run: >
+          pytest --cov scripts scripts/ -vv -rA
+          --junitxml=.git/pytest27.xml
+          --cov-report term-missing
+          --cov-report html:.git/coverage27.html
+          --cov-report xml:.git/coverage27.xml
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v3
```

Comment:
Will this need something to be enabled on this repo for codecov to work? IIRC the app needs to be added to the repo, or has that already be done on an org level?

Reply:
Yes, it will have to be added to the organisation or the repo by an admin in order to be able to make comments.

I can also/alternatively add coverage with coveralls.io, which might be easer to add, maybe.

The workflow also attempts to add a comment to the PR locally on Git if it has permission.

It would be good if I could be added as a member to the https://github.com/xapi-project or the repo at least.

I should be in Cambridge next week, so we'll see, hopefully!

Reply:
if this needs additional permissions on the repo for a 3rdparty app maybe comment out this part of the PR for now, and we can come back to it later.